### PR TITLE
Add s3-storage-wagon plugin for deploying artifacts to s3

### PR DESCRIPTION
**Pull request by frankfu03**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,6 +18,7 @@
         <spring.version>4.3.6.RELEASE</spring.version>
         <aspectj.version>1.8.6</aspectj.version>
         <SONAR_LOGIN>Pass in with -D</SONAR_LOGIN>
+        <gnss.artifactory.repo>gnss-artifactory</gnss.artifactory.repo>
```
Comment (lbodor) [inline pom.xml]: The naming convention for maven properties actually uses a mix of dots and camel case. Camal case is used to separate words, and dots are used to indicate hierarchy.

https://maven.apache.org/pom.html#Properties

gnssArtifactoryRepo

Having said that, you probably don't even need a variable for this. You can just use "s3://gnss-artifactorty/maven-snapots" inline.

Comment (lbodor) [reply pom.xml]: Either way, it's an artifactory, not a repository.

File: pom.xml [hunk 2]
```diff
@@ -305,12 +313,12 @@
     </profiles>
     <distributionManagement>
         <snapshotRepository>
-            <id>nexus.gadevs</id>
-            <url>https://nexus.gadevs.ga/repository/maven-snapshots/</url>
+            <id>igssitelog-bindings-snapshot</id>
+            <url>s3://${gnss.artifactory.repo}/maven-snapshots/${project.artifactId}/${project.version}</url>
```
Comment (lbodor) [inline pom.xml]: Do you need so specify artifact id and project version? I think `mvn deploy` will make this directories for you. If you just end it with `.../maven-snapshots`, `mvn deploy` will use the group id, artifact id, and version to make a directory structure under maven-snapshots.

Comment (frankfu03) [reply pom.xml]: Yes, you are right. I have removed the artifact id and project version from the path. Thanks.

File: pom.xml (hunk 1 above)
Comment (frankfu03) [reply pom.xml]: OK, I removed the property and used "s3://gnss-artifactorty/maven-snapshots" etc.

Comment (lbodor) [reply pom.xml]: :+1:

**Pull request merged**